From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sujeevan Rasaratnam <rsujeevan@gmail.com>
Date: Fri, 24 Jul 2026 11:24:07 -0400
Subject: [PATCH 6/8] Fix pytest 9 warning: passing a non-Collection iterable
 to parametrize is deprecated

pytest 9 emits PytestRemovedIn10Warning for an itertools.product passed
to @pytest.mark.parametrize, which upstream's -Werror turns into a
collection error for tests/test-parallel.py. Wrap it in list().

Upstream-Status: Submitted [https://github.com/h2oai/datatable/pull/3513]

Signed-off-by: Ludovic Henry <git@ludovic.dev>
---
 tests/test-parallel.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/test-parallel.py b/tests/test-parallel.py
index 3d39ec3..7892a64 100644
--- a/tests/test-parallel.py
+++ b/tests/test-parallel.py
@@ -73,10 +73,10 @@ def test_core_progress(testname):
 @skip_on_jenkins
 @pytest.mark.usefixtures("nowin") # `SIGINT` is not supported on Windows
 @pytest.mark.parametrize('parallel_type, nthreads',
-                         itertools.product(
+                         list(itertools.product(
                             [None, "static", "nested", "dynamic", "ordered"],
                             ["1", "half", "all"]
-                         )
+                         ))
                         )
 def test_progress_interrupt(parallel_type, nthreads):
     import signal
-- 
2.43.0

